cmake/modules/zephyr_module.cmake

```diff
@@ -49,7 +49,7 @@ set(cmake_sysbuild_file ${CMAKE_BINARY_DIR}/sysbuild_modules.txt)
 set(zephyr_settings_file ${CMAKE_BINARY_DIR}/zephyr_settings.txt)
 
 if(WEST)
-  set(west_arg "--zephyr-base" ${ZEPHYR_BASE})
+  set(WEST_ARG "--zephyr-base" ${ZEPHYR_BASE})
```

Comment:
`WEST_ARG` is a too generic name which should be considered a local var.
The name `WEST_ARG` give no indication as to where it should be applied, is it perhaps an argument to `west` itself ?

If making this variable global, then it should be renamed into something which more specifically indicates its purpose, for example `ZEPHYR_MODULE_WEST_ARG`.

Alternative, change the code here: https://github.com/zephyrproject-rtos/zephyr/blob/9569e4410475fb62125e769a13222882902d8c75/CMakeLists.txt#L1630
to:
```
    $<$<BOOL:${WEST}>:--zephyr-base=${ZEPHYR_BASE}>
```

Reply:
On second thought, I'm actually wondering why we can't have `-zephyr-base=${ZEPHYR_BASE}` without condition. The script is called with `${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/zephyr_module.py` so `${ZEPHYR_BASE}` must be defined I guess. 

Do you know if it would break anything to remove this condition on `WEST` variable ? 

Looking at `zephyr_module.py` it does not seem it would cause any issue, on the contrary. The following code would make me think, `--zephyr-base` parameter should be mandatory.
 https://github.com/zephyrproject-rtos/zephyr/blob/9569e4410475fb62125e769a13222882902d8c75/scripts/zephyr_module.py#L587-L589

Reply:
> I'm actually wondering why we can't have `-zephyr-base=${ZEPHYR_BASE}` without condition.

Good point, just add this unconditionally.

Code and var name `west_arg` actually originates from before west could be cleanly imported and thus were passed on from CMake to the script. https://github.com/zephyrproject-rtos/zephyr/commit/06365a5ed1d919781d8143bf902e4e3091a49270

later that var seems to just have lived its own life due to the `--zephyr-base` argument.

So please remove the if-test and west arg and just pass on Zephyr base to the script.

Reply:
and suddenly it makes sense why the arg originally was named `west_arg` 😄